PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt

```diff
@@ -327,8 +311,22 @@ open class SparkSubmissionContentPanel(private val myProject: Project, val type:
         toolTipText = "Files to be placed in executor working directory. The path needs to be an Azure Blob Storage Path (path started with wasb://); Multiple paths should be split by semicolon (;) "
     }
 
-    private val referencedFilesTextField: ExpandableTextField = ExpandableTextField(ParametersListUtil.COLON_LINE_PARSER, ParametersListUtil.COLON_LINE_JOINER).apply {
+    private val referencedFilesTextField: TextFieldWithBrowseButton = TextFieldWithBrowseButton(ExpandableTextField(ParametersListUtil.COLON_LINE_PARSER, ParametersListUtil.COLON_LINE_JOINER).apply {
         toolTipText = refFilesPrompt.toolTipText
+    }).apply {
+        textField.document.addDocumentListener(documentValidationListener)
+        button.addActionListener {
+            val root = viewModel.prepareVFSRoot()
+            if (root == null) {
+                StorageChooser.handleInvalidUploadInfo()
+            } else {
+                val chooser = StorageChooser(root){ true }
```

Comment:
```suggestion
                val chooser = StorageChooser(root, StorageChooser.ALL_DIRS_AND_FILES)
```

And add a static member ALL_DIRS_AND_FILES of StorageChooser:
```java
    static public final Condition<VirtualFile> ALL_DIRS_AND_FILES = (vf) -> true;
```